crates/tx5/tests/tests/reconnect.rs

```diff
@@ -1,7 +1,71 @@
 use crate::tests::{
     enable_tracing, ep, ep_with_config, receive_next_message_from, sbd,
+    wait_for_disconnect_message,
 };
 
+#[tokio::test(flavor = "multi_thread")]
+async fn sending_second_message_after_remote_disconnect_succeeds_after_disconnect_event(
+) {
+    enable_tracing();
+
+    let _ = rustls::crypto::aws_lc_rs::default_provider().install_default();
+
+    let sbd_server = sbd().await;
+
+    let (peer_url_1, endpoint_1, mut endpoint_recv_1) = ep(&sbd_server).await;
+    let (peer_url_2, endpoint_2, mut endpoint_recv_2) = ep(&sbd_server).await;
+
+    tracing::error!(?peer_url_1);
+    tracing::error!(?peer_url_2);
+
+    endpoint_1
+        .send(peer_url_2.clone(), b"hello".to_vec())
+        .await
+        .unwrap();
+
+    tokio::time::timeout(std::time::Duration::from_millis(5_000), async {
+        let msg_1 =
+            receive_next_message_from(&mut endpoint_recv_2, peer_url_1.clone())
+                .await;
+        assert_eq!("hello", String::from_utf8_lossy(&msg_1));
+    })
+    .await
+    .unwrap();
+
+    endpoint_2.close(&peer_url_1);
+
```

Comment:
Should we add an assertion here that the message fails to send before the disconnect message is received? Or is it that we do send it but nobody receives it and that's not something you can really test for?

Reply:
See [here](https://github.com/holochain/tx5/blob/0405a23aaf6d94b19195d0e7480467900fd11629/crates/tx5/src/ep.rs#L237-L238).

> The future returned from this method will resolve when the data is handed off to the networking backend or the connection attempt to the peer has failed.

Since the sending side still thinks that it has a connection open to that peer (since it hasn't received the disconnect event yet), there won't be an error but the message won't arrive either. That was actually the intention behind adding this test, to have this behavior be documented somehow...